### ISSUE TITLE
E0110 False Positive with getattr

### ISSUE THREAD

**Issue by EvgeniyMist** [PyCQA/pylint#3789]
### Steps to reproduce
1. Create next files:
mod.py

```
from abc import ABC, abstractmethod


class Cell(ABC):

    @abstractmethod
    def hello(self):
        pass


class SpecCell(Cell):

    def __init__(self, hello_str):
        self.hello_str = hello_str

    def hello(self):
        print(self.hello_str)
```

main.py

```
import mod


SPEC = 'spec'
SpecCell = getattr(mod, '%sCell' % SPEC.capitalize())
obj = SpecCell('Hello world')
obj.hello()
```

2. Run main.py to make sure everything is ok
3. Run `pylint --disable=R,C,W main.py`

### Current behavior
```
main.py:6:6: E0110: Abstract class 'Cell' with abstract methods instantiated (abstract-class-instantiated)
Your code has been rated at 0.00/10
```

### Expected behavior
`Your code has been rated at 10.00/10`
 
### pylint --version output
```
pylint 2.6.0
astroid 2.4.2
Python 3.8.4 (tags/v3.8.4:dfa645a, Jul 13 2020, 16:46:45) [MSC v.1924 64 bit (AMD64)]
```

### Additional information
Ok, there are a lot of ways to work around this problem, for example
- let `SPEC = 'Spec'`, so it is necessary to rid of `.capitalize()`, then everything is ok
- we can just import nessesary class:
`from mod import SpecCell`
and then use globals variables:
`SpecCell = globals()['%sCell' % SPEC.capitalize()] `
and again everything is fine

But this does not change the fact of strange behavior

**Comment by PCManticore** [comment]
There's a bunch of stuff here that pylint is not fully capable of understanding. First we do have support for `getattr` but it won't work with dynamic names. Secondly, we don't have support for string formatting, so `pylint` never knows that the second argument to `getattr` is `SpecCell`. The error is gone though when using `mod.SpecCell` directly so not sure why you need to use `globals()` there. Regardless of the reasoning, we won't be able to support most of these dynamic features since we're still operating in a static analysis realm.
